packages/core/src/browser/node-decoration.ts

```diff
@@ -0,0 +1,355 @@
+                    result.push({ data: caption.substr(i, length), highligh: true });
+                    data = '';
+                    i = i + length - 1;
+                }
+            }
+            if (data.length > 0) {
+                result.push({ data });
+            }
+            if (ranges.length !== 0) {
+                throw new Error('Error occurred when splitting the caption. There was a mismatch between the caption and the corresponding highlighting ranges.');
+            }
+            return result;
+        }
+    }
+    /**
+     * Encapsulates styling information that has to be applied on the node which we decorate.
+     */
+    export interface Data {
+        /**
+         * The higher number has higher priority. If not specified, treated as `0`.
+         * When multiple decorators are available for the same item, and decoration data cannot be merged together,
+         * then the higher priority item will be applied on the decorated element and the lower priority will be ignored.
+         */
+        readonly priority?: number;
+        /**
+         * The font data for the caption.
+         */
+        readonly fontData?: FontData;
+        /**
+         * The background color of the entire row.
+         */
+        readonly backgroundColor?: Color;
+        /**
+         * Optional, leading prefixes right before the caption.
+         */
+        readonly captionPrefixes?: CaptionAffix[];
+        /**
+         * Suffixes that might come after the caption as an additional information.
+         */
+        readonly captionSuffixes?: CaptionAffix[];
+        /**
+         * Optional right-aligned decorations that appear after the node caption and after the caption suffixes (is any).
+         */
+        readonly tailDecorations?: Array<TailDecoration | TailDecorationIcon | TailDecorationIconClass>;
+        /**
+         * Custom tooltip for the decorated item. Tooltip will be appended to the original tooltip, if any.
+         */
+        readonly tooltip?: string;
+        /**
+         * Sets the color of the icon. Ignored if the decorated item has no icon.
+         */
+        readonly iconColor?: Color;
+        /**
+         * Has not effect if given, but the node does not have an associated image.
+         */
+        readonly iconOverlay?: IconOverlay | IconClassOverlay;
+        /**
+         * An array of ranges to highlight the caption.
+         */
+        readonly highlight?: CaptionHighlight;
+    }
+    export namespace Data {
+        /**
+         * Compares the decoration data based on the priority. Lowest priorities come first.
+         */
+        export const comparePriority = (left: Data, right: Data): number => (left.priority || 0) - (right.priority || 0);
+    }
+
+    /**
+     * Specific to tree node decoration.
+     * Tree node that can be decorated explicitly, without the tree decorators.
+     */
+    export interface DecoratedTreeNode extends TreeNode {
```

Comment:
I think this should be excluded from the file.
`node-decoration.ts` is the generic decoration, and should not contain specifics for trees.

Reply:
Sure!